packages/react-components/src/components/tooltip-xy-component.tsx

```diff
@@ -0,0 +1,101 @@
+    }
+
+    render(): React.ReactNode {
+        return (
+            <div
+                ref={this.divRef}
+                onMouseEnter={() => {
+                    if (this.timerId) {
+                        clearTimeout(this.timerId);
+                        this.timerId = undefined;
+                    }
+                }}
+                onMouseLeave={() => {
+                    this.setState({ tooltipData: { opacity: 0 }});
+                }}
+                style={{
+                    padding: 10,
+                    position: 'fixed',
+                    textAlign: 'left',
+                    maxHeight: '250px',
+                    overflow: 'auto',
+                    color: 'white',
+                    border: '1px solid transparent',
+                    backgroundColor: 'rgba(51, 122, 183, 0.9)',
+                    borderRadius: 3,
+                    top: this.state.tooltipData?.top,
+                    left: this.state.tooltipData?.left,
+                    opacity: this.state.tooltipData?.opacity ? this.state.tooltipData.opacity : 0,
+                    zIndex: 999
+                }}
+            >
+                <p style={{margin: '0 0 5px 0'}}>{this.state.tooltipData?.title}</p>
+                <ul style={{padding: '0'}}>
+                    {this.state.tooltipData?.dataPoints?.map((point: { color: string, background: string, label: string, value: string }, i: number) =>
+                        <li key={i} style={{listStyle: 'none', display: 'flex', marginBottom: 5}}>
+                            <div style={{
+                                height: '10px',
+                                width: '10px',
+                                margin: 'auto 0',
+                                border: 'solid thin',
+                                borderColor: point.color,
+                                backgroundColor: point.background
+                                }}
+                            ></div>
+                            <span style={{marginLeft: '5px'}}>{point.label} {point.value}</span>
+                        </li>
+                    )}
+                </ul>
+                {this.state.tooltipData?.zeros > 0 &&
+                    <p style={{marginBottom: 0}}>{this.state.tooltipData?.zeros} other{this.state.tooltipData?.zeros > 1 ? 's' : ''}: 0</p>
+                }
+            </div>
+        );
+    }
+
+    setElement<T>(tooltipData: T): void {
+        if (tooltipData !== this.state.tooltipData && this.state.tooltipData) {
+            if (this.timerId === undefined) {
+                this.timerId = setTimeout(() => {
+                    if (this.state.tooltipData !== tooltipData || this.state.tooltipData === undefined) {
+                        this.setState({ tooltipData: { opacity: 0 }});
+                    }
+                }, 500);
+            }
+        }
+
+        // Delay show
+        setTimeout(() => {
+            this.setState({ tooltipData });
```

Comment:
I'm not sure how it happened, but when hovering around a left point I saw sometimes the tooltip info of a right point.

After a while, it just stopped showing any point (or perhaps always the first point which was y=0?).

Reply:
Solved by changing the way the timeout is used to display the tooltip.